consumer.go

```diff
@@ -33,7 +33,7 @@ func NewConsumer(queue queue.Queue, pool *WorkerPool) *Consumer {
 }
 
 // Start initializes the consumer and starts it, blocking until it is stopped.
-func (c *Consumer) Start() {
+func (c *Consumer) Start() error {
```

Comment:
`Start` is synchronous and only returns when the `WorkerPool` `Close` method returns, which is, when all other goroutines have stopped.

Reply:
I get it, but isn't possible that you call `Start()` which fail but before `Start` we start this _graceful shutdown_ goroutine, what means that it will leak.

Reply:
Done.